### PR TITLE
MAINT: remove redundant open() modes and io.open() alias

### PR DESCRIPTION
**Pull request by mwtoews**
This PR removes redundant [open](https://docs.python.org/3/library/functions.html#open) modes and aliases.

- the default mode for the open builtin is 'r' for read and 't' for text mode, so neither of these need to be specified
- [io.open](https://docs.python.org/3/library/io.html#io.open) is an alias for the open builtin

These edits were mostly automated using [pyupgrade](https://github.com/asottile/pyupgrade/). Happy to keep any exceptions to these edits, please specify.

### CODE REVIEW

File: numpy/_version.py [hunk 1]
```diff
@@ -146,7 +145,7 @@ def git_get_keywords(versionfile_abs):
     # _version.py.
     keywords = {}
     try:
-        with open(versionfile_abs, "r") as fobj:
+        with open(versionfile_abs) as fobj:
```
Comment (seberg) [inline numpy/_version.py:148]: This file is auto-generated, so it seems better not to touch it.

**Comment by charris** [comment]
TBH, I rather like seeing the explicit modes rather than needing to remember them.

**Comment by seberg** [comment]
@charris want to just make a call either way?  I would lean against the `t` (In python 3, binary mode seems rather special to me).  For adding the `'r'` explicitly, I don't care either way.

**Comment by seberg** [comment]
Since it doesn't seem like anyone actually cares much either way going to just put this in.  Thanks @mwtoews.

I think `'t'` is not very necessary to be explicit (and we do it rare enough).  `'r'` may be nice often to be explicit, but Mike isn't wrong about it being obvious in almost all cases either.  And it isn't like it is an unexpected default.

**Pull request merged**